assembly/src/parsers/u32_ops.rs

```diff
@@ -246,7 +246,41 @@ pub fn parse_u32madd(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), As
 /// - u32div.full: 9 cycles
 /// - u32div.unsafe: 1 cycle
 pub fn parse_u32div(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
-    handle_arithmetic_operation(span_ops, op, Operation::U32div, true)
+    let drop_remainder = match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => {
+            assert_u32_operands(span_ops, true);
+            true
+        }
+        2 => match op.parts()[1] {
+            "unsafe" => false,
+            "full" => {
+                assert_u32_operands(span_ops, true);
+                false
+            }
+            _ => {
+                if parse_int_param(op, 1, 0, u32::MAX)? == 0 {
+                    return Err(AssemblyError::invalid_param_with_reason(
+                        op,
+                        1,
+                        "division by 0",
+                    ));
+                }
+
+                assert_u32_and_push_u32_param(span_ops, op, 0)?;
```

Comment:
Let's replace this with `assert_u32_and_push_u32_param(span_ops, op, 1)?;`. I know we'll lose a more specialized error message, but I think it's OK.

Or if we want to keep the more specialized error message, we should probably change this slightly to avoid parsing the immediate parameter twice, and would also probably need to update `parse_u32mod` to follow the same approach for consistency.